process/track/baseBlockTrack.go

```diff
@@ -548,3 +566,90 @@ func (bbt *baseBlockTrack) initNotarizedHeaders(startHeaders map[uint32]data.Hea
 
 	return nil
 }
+
+func (bbt *baseBlockTrack) shouldAddHeader(headerHandler data.HeaderHandler) bool {
+	shardID := headerHandler.GetShardID()
+	if shardID == bbt.shardCoordinator.SelfId() {
+		shardForFirstNotarizedHeader := bbt.shardCoordinator.SelfId()
+		shardForLastNotarizedHeader := bbt.shardCoordinator.SelfId()
+		if bbt.shardCoordinator.SelfId() != core.MetachainShardId {
+			shardForFirstNotarizedHeader = core.MetachainShardId
+		}
+
+		return bbt.shouldAddHeaderForSelfShard(headerHandler, shardForFirstNotarizedHeader, shardForLastNotarizedHeader)
```

Comment:
no need for shardForFirstNoatarized as it is always core.MetaChainShardId.

Reply:
Done